### ISSUE TITLE
Can't access kwarg `assets_folder` of dash_extensions.javascript.Namespace.dump

### ISSUE THREAD

**Issue by araichev** [thedirtyfew/dash-extensions#267]
The above issue caused me a difficult-to-fix permissions error on my server, and i had to result to monkey patching to fix it.
Would be easier if one could somehow configure the `dump`, `assign`, etc. functions to use a custom assets folder path.

**Comment by emilhe** [comment]
What do you think would be the best solution for this feature request? Some kind of auto detection of the assets folder used - or simply an (optional) argument to specify the assets folder?

**Comment by araichev** [comment]
Thanks for your response @emilhe.
Maybe you can follow both approaches.
But i would at least start by using as a default path the assets folder path set by the user in the `DashProxy` object.
That will probably cover most use cases, including mine.

One way to do that is to move the `assets_folder` kwarg out of the `dump` method and into an attribute of the `Namespace` class, make `dump` reference that attribute, and set the attribute's value during instantiation (in line 56) to the assets folder path set in the `DashProxy` object.

Alternatively or additionally, you could give the user finer control of target directories by exposing the `assets_folder` kwarg in the `assign` function.
But that might be overkill.

Hope that helps.
Sorry i'm not familiar enough with the Dash Extension library to convert my suggestion above into a pull request.

**Comment by araichev** [comment]
Yes, but that kwarg is not accessible from the user-facing `assign` function.
Also, it seems more natural for that kwarg to default to the location of the assets folder set when instantiating a `DashProxy` object.
As it stands, if i set a custom assets folder when creating a DashProxy, it's not honoured by the `assign` function.

**Comment by jgaboriaud** [comment]
Any update? I have the same issue.

**Comment by marie-r** [comment]
This is not an ideal workaround, but one can use an enclosure:

```
import dash_extensions.javascript

def js_assignment_function(assets_folder="assets"):
    """
    Generate an a dash extensions assignment function that writes to
    a specific directory.

    :param assets_folder:  path to target assets directory
    :return: function that can be used to assign.

    Example usage:
       js_assign = js_assignment_function(assets_folder="/my/weird/assets")
       # Write myCallback function to the specified assets folder.
       js_callback = js_assign("function(arg1, arg2, ...) { code }", "myCallback")
    """

    # Enclosed function to capture dash_exensions dump function with the
    # specified assets directory.
    def assign(src, name=None):
        """
        Function that will generate assignments, this is returned by
        the enclosing function.  Scoping rules capture the value of
        assets_folder from the enclosing function.

        :param src:  Javascript source code
        :param name: Name for function (provides arbitrary name if not specified)
        :return: handle that can be passed to a dash callback component
        """
        name = dash_extensions.javascript._default_name_space.add(src, name)
        dash_extensions.javascript._default_name_space.dump(assets_folder=assets_folder)
        return dash_extensions.javascript._default_name_space(name)

    return assign
```
It would seem better if DashProxy's assets_folder argument were modified to set the default for the Namespace class, but this does the trick for me.

Regards - Marie
